services/motion/builtin/move_request.go

```diff
@@ -41,78 +44,131 @@ type validatedMotionConfiguration struct {
 	angularDegsPerSec     float64
 }
 
+type requestType uint8
+
+const (
+	requestTypeUnspecified requestType = iota
+	requestTypeMoveOnGlobe
+	requestTypeMoveOnMap
+)
+
 // moveRequest is a structure that contains all the information necessary for to make a move call.
 type moveRequest struct {
+	requestType requestType
+	// origin is only set if requestType == requestTypeMoveOnGlobe
+	origin            spatialmath.GeoPose
+	logger            logging.Logger
 	config            *validatedMotionConfiguration
 	planRequest       *motionplan.PlanRequest
 	seedPlan          motionplan.Plan
 	kinematicBase     kinematicbase.KinematicBase
 	obstacleDetectors map[vision.Service][]resource.Name
 	replanCostFactor  float64
+
+	// from move attempt
```

Comment:
Confused why this is here and we need to store these things?  Wouldnt they be created by the plannerExecutor constructor?

Reply:
discussed offline. We need them for the background polling go routines that trigger replanning